scripts/stats.js

```diff
@@ -0,0 +1,94 @@
+import * as Style from "../src/js/style.js";
+import config from "../src/config.js";
+import { Command } from "commander";
+
+const program = new Command();
+program
+  .option("-a, --all-layers", "summary layer stats")
+  .option("-c, --layer-count", "count number of layers")
+  .option("-s, --layer-size", "size of all layers")
+  .option("-l, --layer <layer id>", "stats about one layer")
+  .option("-loc, --locales <locale1 locale2...>", "language codes", ["mul"])
+  .option("-pp, --pretty", "pretty-print JSON output")
+  .option(
+    "-pg, --print-group <group prefix>",
+    "print a list of the layers in a group"
+  )
+  .option("-pl, --print-layer <layer id>", "print the JSON of a layer");
+program.parse(process.argv);
+
+let opts = program.opts();
+
+if (Object.keys(opts).length === 1) program.help();
+
+if (Object.keys(opts).length === 1) program.help();
+
+let style = Style.build(
+  config.OPENMAPTILES_URL,
+  "https://zelonewolf.github.io/openstreetmap-americana/sprites/sprite",
+  opts.locales
+);
+
+const layers = style.layers;
+const layerCount = layers.length;
+const layerSize = JSON.stringify(layers).length;
+
+const layerMap = new Map();
+const layerGroupMap = new Map();
+const layerSizeStats = new Map();
+const layerGroupSizeStats = new Map();
+const layerGroupCountStats = new Map();
+
+for (let i = 0; i < layers.length; i++) {
+  let layer = layers[i];
+  layerMap.set(layer.id, layers[i]);
+  let layerSize = JSON.stringify(layer).length;
+  let layerGroup = layer.id.split("_", 1)[0];
```

Comment:
It looks like this PR introduces a new naming convention for layer IDs that the statistics code depends on. We should document this naming convention.

Better yet, we should avail ourselves of the `metadata` property, which is an object that you can stuff arbitrary properties into. A style authored in Mapbox Studio inserts `metadata` in abundance, including for grouping layers.

Reply:
Done - contributor's guide updated!

Reply:
Still seems fragile, but let’s ticket out something less fragile as tail work. At a glance, we could group the layers by source layer to get pretty similar statistics, modulo some layer organization weirdness in OpenMapTiles.